src/g3pylib/recordings/recording.py

```diff
@@ -171,40 +174,121 @@ def uuid(self) -> str:
-                f"Could not retrieve file name for gaze data from recording data collected from {data_url}."
-            )
-            raise InvalidResponseError
+        gaze_file_name = self._get_file_name_from_json_data(
+            await self.get_json_data(), "gaze", data_url
+        )
         return f"{data_url}/{gaze_file_name}?use-content-encoding=true"
+
+    async def download_files(self, path: str = ".") -> str:
+        """
+        Downloads all recording files into one folder under the specified `path`.
+        Returns name of the downloaded folder under `path`.
+        """
+        data_url = f"{self._http_url}{await self.get_http_path()}"
+        data_str = await self.get_json_data()
+        data_json = json.loads(data_str)
+
+        # workaround for making subsequent aiohttp clientsessions
+        # TODO: find a more appropriate solution
+        await asyncio.sleep(0.5)
+
+        # create download folder and the meta folder within it
+        folder_name = data_json["name"]
+        os.makedirs(os.path.join(path, folder_name), exist_ok=True)
+        meta_folder_name = data_json["meta-folder"]
+        os.makedirs(os.path.join(path, folder_name, meta_folder_name), exist_ok=True)
+
+        # write json data to recording.g3 file
+        with open(os.path.join(path, folder_name, "recording.g3"), "w") as f:
+            f.write(data_str)
+
+        # generate filenames and file urls
+        scenevideo_file_name = self._get_file_name_from_json_data(
+            data_str, "scenecamera", data_url
+        )
+        scenevideo_url = f"{data_url}/{scenevideo_file_name}"
+
+        gazedata_file_name = self._get_file_name_from_json_data(
+            data_str, "gaze", data_url
+        )
+        gazedata_url = f"{data_url}/{gazedata_file_name}"
+
+        eventdata_file_name = self._get_file_name_from_json_data(
+            data_str, "events", data_url
+        )
+        eventdata_url = f"{data_url}/{eventdata_file_name}"
+
+        imudata_file_name = self._get_file_name_from_json_data(
+            data_str, "imu", data_url
+        )
+        imudata_url = f"{data_url}/{imudata_file_name}"
+
+        async with aiohttp.ClientSession() as session:
+
+            async def download(url: str, file_name: str) -> None:
+                async with session.get(url) as response:
+                    with open(os.path.join(path, folder_name, file_name), "wb") as f:
+                        f.write(await response.read())
+
+            async def download_meta(key: str) -> None:
+                with open(
+                    os.path.join(path, folder_name, meta_folder_name, key), "w"
+                ) as f:
+                    encoded_value = await self.meta_lookup(key)
+                    # decode returned string if it is base64 encoded, otherwise write it raw
+                    try:
+                        value = base64.b64decode(encoded_value).decode()
+                    except ValueError:
```

Comment:
I found that the default metadata like `HuSerial`, `RuSerial` and `RuVersion` are base64-encoded as described in the developer guide, but custom ones, like the ones `tests/api_components/test_recording.py::test_meta_data` created, are plain text. Is this intended behavior?